### PR TITLE
DOC: Fix #12050 by updating numpy.random.hypergeometric docs

### LINKED ISSUE #12050: Documentation for numpy.random.hypergeometric is confusing and, in places, incorrect
It is unclear from the [documentation](https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.random.hypergeometric.html) whether numpy.random.hypergeometric is returning the number of "good" or "bad" elements in the sample. Testing shows it's the number of good elements, but it would be nice if the description of **out** made this explicit. Adding to the confusion, I believe the formula in the notes is incorrect (the combination in the upper left should be "m choose x"), but this then becomes the probability of choosing "x" bad elements. Changing the description to "x = number of good elements in sample", "n = nsample", "m = ngood", and "N = total number of elements (ngood + nbad)" would remove the discrepancy between code and notes.

### PR DESCRIPTION
**Pull request by kshyatt**
- Add backticks for some parameters in the body
- Address the wording changes in the issue

#12050

### CODE REVIEW

**Comment by eric-wieser** [comment]
A tip - if you don't mention the github issue in the body of the PR (i.e. not the title), then github won't link the two together. I'll edit your post to add it.

File: numpy/random/mtrand/mtrand.pyx [hunk 1]
```diff
@@ -4203,9 +4203,9 @@ cdef class RandomState:
         Draw samples from a Hypergeometric distribution.
 
         Samples are drawn from a hypergeometric distribution with specified
-        parameters, ngood (ways to make a good selection), nbad (ways to make
-        a bad selection), and nsample = number of items sampled, which is less
-        than or equal to the sum ngood + nbad.
+        parameters, ``ngood`` (ways to make a good selection), ``nbad`` (ways to make
+        a bad selection), and ``nsample`` = number of items sampled, which is less
```
Comment (eric-wieser) [inline numpy/random/mtrand/mtrand.pyx]: nit: replace `= description` with `(description)` to match the other two descriptions here

**Comment by mattip** [comment]
The double back-ticks should all be single-backticks for function parameters. These are then rendered specially, sphinx "knows" they are parameters. Otherwise looks good

File: numpy/random/mtrand/mtrand.pyx [hunk 2]
```diff
@@ -4226,7 +4226,7 @@ cdef class RandomState:
         Returns
         -------
         out : ndarray or scalar
-            Drawn samples from the parameterized hypergeometric distribution.
+            Number of drawn good samples from the parameterized hypergeometric distribution.
```
Comment (eric-wieser) [inline numpy/random/mtrand/mtrand.pyx]: Hmm, I don't think this helps. There are two different samples being described here - `hypergeometric(..., size=N)` returns `N` samples drawn from a hypergeometric distribution.

But the act of _"drawing from a hypergeometric distribution"_ means _"draw samples from (the number of good samples from `nsample` draws without replacement of `[good] * ngood + [bad] * nbad`)"_.

My example is a pretty lousy way to phrase it, but I think it shows what confusion needs to be resolved here.

Comment (eric-wieser) [reply numpy/random/mtrand/mtrand.pyx]: Perhaps instead, keep the original docstring, and add

> Each sample is the number of good items within a randomly selected subset of size `nsample` taken from a set of `ngood` good items and `nbad` bad items.

**Comment by kshyatt** [comment]
Thanks @mattip for tip about backticks! And thank you @eric-wieser for fixing my PR body. I'll fix up the docstring.

**Comment by mattip** [comment]
Thanks @kshyatt

**Pull request merged**